src/models/book/dtb/segment.coffee

```diff
@@ -167,91 +243,31 @@ class LYT.Segment
-            continue
-          else
-            # Finally just give up and use naturalHeight/naturalWidth
-            attr = type.replace /^([a-z])/g, (m, p1) -> 'natural' + p1.toUpperCase()
-            log.warn "render.content: imageDim: no #{type} attribute or css #{type} on image. Falling back to #{attr} which is not known to be cross browser"
-            result[type] = image[attr]
-        result
-
-      getCanvasScale = (canvasSize, imageSize) ->
-        # TODO: We should check that imageSize.width / canvasSize.width == imageSize.height / canvasSize.height
-        #       ...and complain loudly if it isn't the case
-        if canvasSize.width != imageSize.width
-          try
-            imageSize.width / canvasSize.width
-          catch e
-            1
-        else
-          return 1
-
-      loadImage = (image) =>
-        log.message "Segment: #{this.url()}: parseContent: initiate preload of image #{image.src}"
-        # Note that clearing the timeout has to be done as the first thing in
-        # both handlers. We still have a race condition where the timer may
-        # fire just before being cleared.
-        errorHandler = (event) ->
-          clearTimeout image.timer
-          if image.attempts-- > 0
-            backoff = Math.ceil(Math.exp(LYT.config.segment.imagePreload.attempts - image.attempts + 1) * 50)
-            log.message "Segment: parseContent: preloading image #{image.src} failed, #{image.attempts} attempts left. Waiting for #{backoff} ms."
-            doLoad = -> loadImage image
-            setTimeout doLoad, backoff
-          else
-            log.error "Segment: parseContent: unable to preload image #{image.src}"
-            image.deferred.reject image, event
-
-        doneHandler = (event) ->
-          clearTimeout image.timer
-          log.message "Segment: parseContent: loaded image #{image.src}"
-          image.deferred.resolve image, event
-
-        # Set timeout, so we can retry again if the load stalls
-        image.timer = setTimeout errorHandler, LYT.config.segment.imagePreload.timeout
-        # 1998 called; they want their preloading technique back
-        tmp = new Image()
-        $(tmp).load(doneHandler).error(errorHandler)
-        tmp.src = image.src
 
       if image.length != 1
         log.error "Segment: parseContent: can't create reliable cartoon type " +
           "with multiple or zero images: #{this.url()}"
         throw "Segment: parseContent: unable to find exactly one image in " +
           "cartoon display div"
 
-      # FIXME: We're basically cloning the image and div elements. This should
-      # be fixed when we move to a better rendering architecture
-      @image = image.clone().wrap('<p>').parent().html()
-      @div = sourceContent.clone().wrap('<p>').parent().html()
-      @canvasSize = getCanvasSize image
+      @areaData = @parseArea sourceContent
+      @cartoonImage = image
+
+      @canvasSize = @getCanvasSize image
```

Comment:
Antager at denne ændring betyder at vi undgår image cloning fra nu af?

Reply:
It's partly to avoid cloning of the image, but mostly it's to load the image only once.

`Img[data-src]` means the image isn't loaded yet, so `Img[src]` is set to `Img[data-src]` when we want to load it. The clone didn't get this change and the browser might request the image again when the image is used.